component/attr_cache/attr_cache.go

```diff
@@ -260,35 +227,23 @@ func getParentDir(childPath string) string {
 // Do not use this with ac.cacheDirs set
 func (ac *AttrCache) invalidateDirectory(path string) {
 	// Invalidate all descendants of the path, then invalidate the path
-	// For example, filesystem: a/, a/b, a/c, aa/, ab.
-	// When we invalidate directory a, we only want to invalidate a/, a/b, and a/c.
-	// If we do not conditionally extend a, we would accidentally invalidate aa/ and ab
-
-	// Add a trailing / so that we only invalidate child paths under the directory and not paths that have the same prefix
-	prefix := dirToPrefix(path)
-
-	for key, value := range ac.cacheMap {
-		if strings.HasPrefix(key, prefix) {
-			// don't invalidate directories when cacheDirs is true
-			if ac.cacheDirs && value.attr.IsDir() {
-				continue
-			}
-			value.invalidate()
-		}
-	}
 
-	// We need to invalidate the path itself since we only handle children above.
-	if !ac.cacheDirs {
-		ac.invalidatePath(path)
+	toBeInvalid, err := ac.cacheMap.get(path)
+	if err != nil {
+		log.Err("could not invalidate cached attr item due to the following error: ", err)
+		return
 	}
-}
-
-// invalidatePath: invalidates a path
-func (ac *AttrCache) invalidatePath(path string) {
-	// Keys in the cache map do not contain trailing /, truncate the path before referencing a key in the map.
-	value, found := ac.cacheMap[internal.TruncateDirName(path)]
-	if found {
-		value.invalidate()
+	// don't invalidate directories when cacheDirs is true
+	if !ac.cacheDirs && toBeInvalid.attr.IsDir() {
```

Comment:
Discussion about this block is here:
https://github.com/Seagate/cloudfuse/pull/15#discussion_r1365889372